ts/parsers/slideParser.ts

```diff
@@ -1,14 +1,141 @@
+                    "http://schemas.openxmlformats.org/officeDocument/2006/relationships/slideLayout"
+                ) {
+                    layoutFilename = relationship["$"]["Target"].replace("../", "ppt/");
+                    break;
+                }
+            }
+        } else {
+            layoutFilename = relationshipArray["$"]["Target"].replace("../", "ppt/");
+        }
+        // Open slideLayoutXX.xml
+        const slideLayoutContent = await ZipHandler.parseSlideAttributes(layoutFilename);
+
+        // Read slide master filename of the slidelayout (Get slideMasterXX.xml)
+        // @resName: ppt/slideLayouts/slideLayout1.xml
+        // @masterName: ppt/slideLayouts/_rels/slideLayout1.xml.rels
+        const slideLayoutResFilename =
+            layoutFilename.replace("slideLayouts/slideLayout", "slideLayouts/_rels/slideLayout") +
+            ".rels";
+        const slideLayoutResContent = await ZipHandler.parseSlideAttributes(slideLayoutResFilename);
+        relationshipArray = slideLayoutResContent["Relationships"]["Relationship"];
+        let masterFilename = "";
+        if (relationshipArray.constructor === Array) {
+            for (const relationship of relationshipArray) {
+                if (
+                    relationship["$"]["Type"] ===
+                    "http://schemas.openxmlformats.org/officeDocument/2006/relationships/slideMaster"
+                ) {
+                    masterFilename = relationship["$"]["Target"].replace("../", "ppt/");
+                    break;
+                }
+            }
+        } else {
+            masterFilename = relationshipArray["$"]["Target"].replace("../", "ppt/");
+        }
+        // Open slideMasterXX.xml
+        const slideMasterContent = await ZipHandler.parseSlideAttributes(masterFilename);
+
+        return {
+            slideLayoutTables: this.indexNodes(slideLayoutContent),
+            slideMasterTables: this.indexNodes(slideMasterContent)
+        };
+    }
+
+    public static indexNodes(content) {
+        try {
+            const keys = Object.keys(content);
+            const spTreeNode = content[keys[0]]["p:cSld"][0]["p:spTree"][0];
+
+            const idTable = {};
+            const idxTable = {};
+            const typeTable = {};
+
+            for (const key in spTreeNode) {
+                if (key !== "p:sp") {
+                    continue;
+                }
+
+                var targetNode = spTreeNode[key];
+
+                if (targetNode.constructor === Array) {
+                    for (const node of targetNode) {
+                        const nvSpPrNode = node["p:nvSpPr"];
+                        const id = getAttributeByPath(nvSpPrNode[0], ["p:cNvPr", "$", "id"]);
+                        const idx = getAttributeByPath(nvSpPrNode[0], [
+                            "p:nvPr",
+                            "p:ph",
+                            "$",
+                            "idx"
```

Comment:
there appears to be some inconsistency, sometimes these elements are in line and sometimes line break after each .. just check and make them consistent please

Reply:
some formatting got off, most likely when i worked with on the mac, and settings did not sync. i have reformatted all the files and have set the print width to 120. 
I have changed it in the vscode settings and have pushed it. so it doesn't happen again